src/cx/cx.ts

```diff
@@ -1,456 +1,483 @@
+        } catch (e) {
+            throw new Error("Error running kics remediation");
+        }
+        return [kics, process];
+
+    }
+
+    async learnMore(queryID: string): Promise<CxLearnMoreDescriptions[] | undefined> {
+        let r = [];
+        const config = this.getAstConfiguration();
+        if (!config) {
+            return [];
+        }
+        const cx = new CxWrapper(config);
+        const scans = await cx.learnMore(queryID);
+        if (scans.payload && scans.exitCode === 0) {
+            r = scans.payload;
+        } else {
+            throw new Error(scans.status);
+        }
+        return r;
+    }
+
+    updateStatusBarItem(text: string, show: boolean, statusBarItem: vscode.StatusBarItem) {
+        statusBarItem.text = text;
+        show ? statusBarItem.show() : statusBarItem.hide();
+    }
+
+    async installAsca(): Promise<CxAsca> {
+        let config = this.getAstConfiguration();
+        if (!config) {
+            config = new CxConfig();
+        }
+        const cx = new CxWrapper(config);
+        const scans = await cx.scanAsca(null, true, constants.vsCodeAgent);
+        if (scans.payload && scans.exitCode === 0) {
+            return scans.payload[0];
+        } else {
+            return this.getAscaError(scans.status, "Failed to run ASCA engine");
+        }
+    }
+
+    private getAscaError(scanStatus: string, errorMessage: string) {
+        console.error(errorMessage);
+        const errorRes = new CxAsca();
+        errorRes.error = scanStatus;
+        return errorRes;
+    }
+
+    async scanAsca(sourcePath: string): Promise<CxAsca> {
+        let config = this.getAstConfiguration();
+        if (!config) {
+            config = new CxConfig();
+        }
+        const cx = new CxWrapper(config);
+        const scans = await cx.scanAsca(sourcePath, false, constants.vsCodeAgent);
+        if (scans.payload && scans.exitCode === 0) {
+            return scans.payload[0];
+        } else {
+            return this.getAscaError(scans.status, "Fail to call ASCA scan");
+        }
+    }
+
+    async authValidate(logs: Logs): Promise<boolean> {
+        const invalidAPIKeyMsg = "Failed to authenticate to Checkmarx One server";
+        const authSuccessMsg = "Successfully authenticated to Checkmarx One server";
+        const config = this.getAstConfiguration();
+        const cx = new CxWrapper(config);
+        try {
+            const valid = await cx.authValidate();
+            if (valid.exitCode === 0) {
+                logs.info(authSuccessMsg);
+                vscode.window.showInformationMessage(authSuccessMsg);
+                return true;
+            } else {
+                logs.error(`${invalidAPIKeyMsg}: ${valid.status}`);
+                vscode.window.showErrorMessage(invalidAPIKeyMsg);
+                return false;
+            }
+        } catch (error) {
+            logs.error(`${invalidAPIKeyMsg}: ${error}`);
```

Comment:
if we got exit code != 0, it means that apikey is invalid
but if we catch an exception it means other issue happened, could be not related to apikey itself
no?